x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/TransformIndexerStats.java

```diff
@@ -214,6 +249,19 @@ public double getExpAvgDocumentsProcessed() {
         return expAvgDocumentsProcessed;
     }
 
+    public void incrementNumDeletedDocuments(long n) {
+        assert (n >= 0);
+        numDeletedDocuments += n;
+    }
+
+    public void markStartDelete() {
+        startDeleteTime = System.nanoTime();
+    }
+
+    public void markEndDelete() {
+        deleteTime += ((System.nanoTime() - startDeleteTime) / 1000000);
```

Comment:
I implemented it the same way as the others, see:
https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexing/IndexerJobStats.java#L172

I don't want to do it differently than the other places. I agree about using time instead of low-level types, I might change it in _all_ places as a follow up.